input/blog/2020-07-12-article-blazor-compelling-example.md

```diff
@@ -0,0 +1,313 @@
+﻿---
+title: ReactiveUI On The Web with Blazor
+category: 
+  - Article
+author: Rich Bryant
+---
+
+<img src="https://avatars0.githubusercontent.com/u/2327577?s=200&v=4" align="right" style="height:8em"/>
+
+# ReactiveUI On The Web With Blazor!  
+## It really works, honest.  
+  
+I hope by now people are starting to learn about the existence of the [ReactiveUI.Blazor](https://www.nuget.org/packages/ReactiveUI.Blazor/) package.  Blazor means "writing Single Page Apps like Angular or Vue in C#" and it has all the potential in the world.  And ReactiveUI supports it.  The official [samples are here](https://github.com/reactiveui/ReactiveUI.Samples/tree/main/blazor) but I don't feel that they really sell the reason _why_ you'd want to do web work with ReactiveUI, and I should know, I wrote them.
+  
+For me, the big seller is that you've already got a WPF or Xamarin or WinForms app using ReactiveUI and you want a web version.  In such a case, you can rewrite everything from scratch _or_ you can use Blazor and keep the logic in your ViewModels absolutely unchanged.  Sounds nice in theory but the web is a tricky thing, so does it actually work?  I decided to try it.  
+  
+### The ReactiveUI Compelling Example  
+  
+The ReactiiveUI website Getting Started section links a [Compelling Example](https://www.reactiveui.net/docs/getting-started/compelling-example), a simple Nuget browser that you can build with copypasta in about 5 minutes.  
+
+<img src="https://i.imgur.com/GoczlPS.png" align="center"/>
+
+As you can see, it has text input, it does pipelines, it has data, it has views and viewmodels.... Why not, I thought? What are the steps?  
+  
+1. Follow the instructions Rich, you idiot
+
+Yeah, that makes sense.  Step 1 - build the Compelling Example just like the instructions say except that I used WPF in .NET Core 3.1 which worked flawlessly and I didn't have to change anything.  I mostly did this because Blazor is a .NET Core app and I wanted to avoid any hassles later.  The screenshot above is my working WPF project.
+
+2. Add a Blazor project.  
+
+I used Blazor AspNetCore Hosted for reasons which I hope will become clear later and I moved the actual "get the data" stuff to a service inside the Server project.  I also created a ViewModels project (a .NET Standard 2.1 clas library) because I don't want to copy my ViewModels, I want them shared.  
```

Comment:
a .NET Standard 2.1 class library

Reply:
Will fix that in the morning. Its 0120hrs here.